synapse/replication/http/devices.py

```diff
@@ -78,5 +79,71 @@ async def _handle_request(  # type: ignore[override]
         return 200, user_devices
 
 
+class ReplicationUploadKeysForUserRestServlet(ReplicationEndpoint):
+    """Ask master to upload keys for the user and send them out over federation to
+    update other servers.
+
+    This must happen on master so that the results can be correctly cached in
+    the database and streamed to workers.( Is this accurate?)
```

Comment:
```suggestion
    For now, only the master is permitted to handle key upload requests;
    any worker can handle key query requests (since they're read-only).
```